### ISSUE TITLE
Produces/Consumes media type does not support parameters (rfc6838)

### ISSUE THREAD

**Issue by kmudrick** [twilio/guardrail#796]
In a spec, specifying `produces` and `consumes` with media type parameters generates broken code (does not compile).

Example:
```
consumes:
  - application/json;foo=bar

produces:
  - application/json;baz=bat
```

I am guessing [that the `ApplicationJson` check](https://github.com/twilio/guardrail/blob/master/modules/codegen/src/main/scala/com/twilio/guardrail/generators/Scala/Http4sHelper.scala#L64-L97) could maybe be changed to use `exists` with a predicate that matches against the start of the string?  I can open a PR if need be to illustrate/fix.

**Comment by kelnos** [comment]
I don't think we considered that people might put parameters in their media types, which I guess is valid according to the spec.

The right way to fix this would be to augment `ContentType` to match against whatever is before the `;`.  We should also store the parameters so we can make use of them in the generators, but as a first step I'm fine with just fixing the matching to not account for parameters at all.
